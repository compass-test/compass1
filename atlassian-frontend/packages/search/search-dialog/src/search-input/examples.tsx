import React, { useState } from 'react';
import { action } from '@storybook/addon-actions';
import SearchInput from './search-input';
import { ThemeProvider } from '../theme';
import { generateTheme } from '@atlaskit/atlassian-navigation';

const commonProps = {
  placeholder: 'Search',
  onClick: action('onClick'),
  onInput: action('onInput'),
};

const themes = [
  {
    name: 'huge',
    backgroundColor: '#FFFFFF',
    highlightColor: '#D8388A',
  },
  {
    name: 'showpo',
    backgroundColor: '#E8CBD2',
    highlightColor: '#333333',
  },
  {
    name: 'up',
    backgroundColor: '#EF816B',
    highlightColor: '#FDEE80',
  },
  {
    name: '86400',
    backgroundColor: '#000448',
    highlightColor: '#6FF2B4',
  },
  {
    name: 'netflix',
    backgroundColor: '#272727',
    highlightColor: '#E94E34',
  },
];

export const CollapsedWithNoText = () => (
  <SearchInput {...commonProps} isExpanded={false} value="" />
);

export const ExpandedWithNoText = () => (
  <SearchInput {...commonProps} isExpanded value="" />
);

export const CollapsedWithText = () => (
  <SearchInput
    {...commonProps}
    isExpanded={false}
    value="some random search query"
  />
);

export const ExpandedWithText = () => (
  <SearchInput {...commonProps} isExpanded value="some random search query" />
);

export const WithTheming = () => (
  <div>
    {[true, false].map((isExpanded) =>
      themes.map((theme) => {
        return (
          <div
            key={theme.name}
            style={{ backgroundColor: theme.backgroundColor, padding: '5px' }}
          >
            <ThemeProvider partialSearchCSS={generateTheme(theme).mode.search}>
              <SearchInput
                {...commonProps}
                isExpanded={isExpanded}
                value={isExpanded ? theme.name : ''}
              />
            </ThemeProvider>
          </div>
        );
      }),
    )}
  </div>
);

export const WithTooltip = () => (
  <div>
    <SearchInput
      {...commonProps}
      isExpanded={false}
      value=""
      tooltipContent={<p>I am a tooltip</p>}
    />

    <br />
    <br />

    <SearchInput
      {...commonProps}
      isExpanded
      value=""
      tooltipContent={<p>I am a tooltip</p>}
    />
  </div>
);

export const Interactive = () => {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  const [value, setValue] = useState('');
  return (
    <SearchInput {...commonProps} isExpanded value={value} onInput={setValue} />
  );
};

export default { title: 'Search Dialog/Search Input' };
