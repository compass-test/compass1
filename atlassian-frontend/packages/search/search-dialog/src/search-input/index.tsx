// Types
export type { Props as SearchInputProps } from './search-input';
export { default as SearchInput } from './search-input';
