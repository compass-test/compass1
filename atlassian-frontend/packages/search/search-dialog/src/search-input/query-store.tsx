export interface InputSkeletonState {
  query: string;
  placeholder: string;
  isFocused: boolean;
  selectionStart: number;
  selectionEnd: number;
  selectionDirection: 'forward' | 'backward' | 'none';
  listeners: SSREventListener[];
}

export type SSREventListener<
  TNode = Node,
  TEventName = EventName,
  TEventListener = EventListener
> = [TNode, TEventName, TEventListener];

export type EventName = keyof HTMLElementEventMap;

export type EventListener = <T extends EventName>(
  this: HTMLElement,
  ev: HTMLElementEventMap[T],
) => any;
declare global {
  interface Window {
    inputSkeletonState?: InputSkeletonState;
  }
}

interface getInputSkeletonSelection {
  selectionStart: number;
  selectionEnd: number;
  selectionDirection: 'forward' | 'backward' | 'none';
}

const getInputSkeletonQuery = (): string => {
  return window?.inputSkeletonState?.query || '';
};

const setInputSkeletonQuery = (newQuery: string): void => {
  if (window?.inputSkeletonState) {
    window.inputSkeletonState.query = newQuery;
  }
};

const getInputSkeletonFocus = (): boolean => {
  return window?.inputSkeletonState?.isFocused || false;
};

const setInputSkeletonIsFocus = (newIsFocused: boolean): void => {
  if (window?.inputSkeletonState) {
    window.inputSkeletonState.isFocused = newIsFocused;
  }
};

const getInputSkeletonSelection = (): getInputSkeletonSelection => {
  const selectionStart = window?.inputSkeletonState?.selectionStart || 0;
  const selectionEnd = window?.inputSkeletonState?.selectionEnd || 0;
  const selectionDirection =
    window?.inputSkeletonState?.selectionDirection || 'none';

  return { selectionStart, selectionEnd, selectionDirection };
};

const setInputSkeletonSelection = (
  newSelectionStart: number,
  newSelectionEnd: number,
  newSelectionDirection: 'forward' | 'backward' | 'none',
): void => {
  if (window?.inputSkeletonState) {
    window.inputSkeletonState.selectionStart = newSelectionStart;
    window.inputSkeletonState.selectionEnd = newSelectionEnd;
    window.inputSkeletonState.selectionDirection = newSelectionDirection;
  }
};

export {
  getInputSkeletonQuery,
  setInputSkeletonQuery,
  getInputSkeletonFocus,
  setInputSkeletonIsFocus,
  getInputSkeletonSelection,
  setInputSkeletonSelection,
};
