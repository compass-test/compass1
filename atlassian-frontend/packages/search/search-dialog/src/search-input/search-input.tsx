import React from 'react';
import Tooltip from '@atlaskit/tooltip';
import { IconButton, SearchCSS } from '@atlaskit/atlassian-navigation';
import ArrowLeftIcon from '@atlaskit/icon/glyph/arrow-left';
import SelectClearIcon from '@atlaskit/icon/glyph/select-clear';
import { useTheme } from '../theme';
import { SearchIcon, LargeSearchIcon } from '../icons';
import {
  MobileSearchIconWrapper,
  DesktopSearchIconWrapper,
  SearchInputContainer,
  SearchInputField,
  ArrowIconWrapper,
  ClearIconWrapper,
  HiddenTooltip,
} from './search-input.styled';

export interface Props {
  /**
   * Whether the search input is currently in the expanded state or not.
   */
  isExpanded: boolean;
  /**
   * Callback that is called when the value in the search input changes
   */
  onInput?: (value: string) => void;
  /**
   * Callback that is called when the input is clicked
   */
  onClick?: (e?: React.MouseEvent<HTMLElement>) => void;
  /**
   * Callback that is called when the user pressed enter while focused on this input
   */
  onEnter?: (e: React.KeyboardEvent) => void;
  /**
   * Callback that is called when the user clicks the \`<-\` (back button). This button is only available when the screen is below a certain width (mobile mode).
   */
  onBack?: (e: React.MouseEvent<HTMLElement>) => void;
  /**
   * Placeholder text for the input
   */
  placeholder?: string;
  /**
   * Tooltip text for the input. The tooltip will appear when the user hovers over the input.
   */
  tooltipContent?: JSX.Element;
  /**
   * Value of the search input, if this provided then the component is considered controlled.
   */
  value?: string | number | string[];
  /**
   * Whether or not to show the x button that clears the input
   */
  showClearButton?: boolean;
  /**
   * Callback that is called to clear the search input.
   */
  onClear?: () => void;
  /**
   * Callback to set whether or not to allow the dialog to change expansion states
   */
  allowChangeExpand?: (expand: boolean) => void;
  /**
   * Label text for the Exit Search button
   */
  exitSearchLabel?: string;
  /**
   * Label text for the Clear Search button
   */
  clearSearchLabel?: string;
}

interface InternalProps {
  forwardRef?: React.Ref<HTMLInputElement>;
  theme: SearchCSS;
}

const MAX_INPUT_LENGTH = 500;

export const SearchInput: React.FunctionComponent<Props & InternalProps> = ({
  onInput,
  onClick,
  onEnter,
  isExpanded,
  placeholder,
  forwardRef,
  value,
  theme,
  tooltipContent,
  onBack,
  showClearButton,
  onClear,
  allowChangeExpand,
  exitSearchLabel,
  clearSearchLabel,
}) => {
  const handleInput = (e: React.FormEvent<HTMLInputElement>) => {
    if (onInput) {
      onInput(e.currentTarget.value);
    }
  };

  const handleClick = (e: React.MouseEvent<HTMLElement> | undefined) => {
    if (onClick) {
      onClick(e);
    }
  };

  const handleTabEnter = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter' && onClick) {
      onClick();
    }
  };

  const handleKeyDown = (e: React.KeyboardEvent) => {
    if (e.nativeEvent?.isComposing || e.keyCode === 229) {
      return;
    }
    if (!e.defaultPrevented && e.key === 'Enter') {
      e.preventDefault();
      onEnter && onEnter(e);
    }
  };

  const handleClear = () => {
    onClear?.();
  };

  const TooltipComponent =
    tooltipContent && !isExpanded ? undefined : HiddenTooltip;

  return (
    <Tooltip
      content={tooltipContent}
      component={TooltipComponent}
      position="bottom"
    >
      <SearchInputContainer {...theme} isExpanded={isExpanded}>
        <MobileSearchIconWrapper isExpanded={isExpanded}>
          <IconButton
            onClick={handleClick}
            tooltip={null}
            icon={<LargeSearchIcon />}
          />
        </MobileSearchIconWrapper>
        <DesktopSearchIconWrapper
          {...theme}
          isExpanded={isExpanded}
          onClick={handleClick}
          onKeyPress={handleTabEnter}
          tabIndex={0}
        >
          <SearchIcon />
        </DesktopSearchIconWrapper>
        {isExpanded && (
          <ArrowIconWrapper {...theme} onClick={onBack}>
            <ArrowLeftIcon
              label={exitSearchLabel || 'Exit Search'}
              size="large"
              primaryColor={theme.default.color}
            />
          </ArrowIconWrapper>
        )}
        <SearchInputField
          data-test-id="search-dialog-input"
          onInput={handleInput}
          onChange={handleInput}
          onKeyDown={handleKeyDown}
          onClick={handleClick}
          isExpanded={isExpanded}
          placeholder={placeholder}
          ref={forwardRef || undefined}
          value={value}
          maxLength={MAX_INPUT_LENGTH}
          {...theme}
        />
        {showClearButton && (
          <ClearIconWrapper
            {...theme}
            onClick={handleClear}
            onMouseEnter={() => {
              allowChangeExpand?.(false);
            }}
            onMouseLeave={() => {
              allowChangeExpand?.(true);
            }}
            tabIndex={0}
          >
            <SelectClearIcon
              label={clearSearchLabel || 'Clear search session'}
              size="small"
            />
          </ClearIconWrapper>
        )}
      </SearchInputContainer>
    </Tooltip>
  );
};

export default React.forwardRef<HTMLInputElement, Props>((props, ref) => {
  const theme = useTheme();
  return <SearchInput {...props} theme={theme} forwardRef={ref} />;
});
