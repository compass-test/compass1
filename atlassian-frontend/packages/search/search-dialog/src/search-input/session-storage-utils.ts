export const STICKY_SEARCH_KEY = 'atlassian.search-dialog-sticky';
export const SEARCH_QUERY_KEY = 'atlassian.search-dialog-query';
export const SEARCH_FILTERS_KEY = 'atlassian.search-dialog-filters';

export const getSessionItem = (key: string) => {
  return window.sessionStorage.getItem(key);
};

export const setSessionItem = (key: string, value: string) => {
  try {
    window.sessionStorage.setItem(key, value);
  } catch (error) {
    if (error.code === DOMException.QUOTA_EXCEEDED_ERR) {
      // Do nothing if the sessionStorage capacity is reached
    } else {
    }
  }
};

export const deleteSessionItem = (key: string) => {
  window.sessionStorage.removeItem(key);
};
