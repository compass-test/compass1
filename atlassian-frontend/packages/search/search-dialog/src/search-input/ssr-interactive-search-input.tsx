/* eslint-disable import/extensions */
import React, { FC, useRef, useLayoutEffect } from 'react';
import {
  SSRsearchInputField,
  SSRsearchInputContainer,
  SSRdesktopSearchIconWrapper,
  SSRmobileSearchIconWrapper,
  SSRformContainer,
  SSRsearchInputWrapper,
} from './search-input.styled';
// @ts-ignore
import { script } from './ssr-scripts/min-script-interactive-search-input';

import { SearchIcon, LargeSearchIcon } from '../icons';

export interface Props {
  advancedSearchUrl?: string;
  placeholder?: string;
}
export const SSRsearchInput: FC<Props> = ({
  advancedSearchUrl,
  placeholder,
}) => {
  const scriptRef = useRef(null);

  useLayoutEffect(() => {
    const current: any = scriptRef?.current;
    if (current) {
      // eslint-disable-next-line
      eval(current.innerHTML);
    }
  });

  return (
    <SSRsearchInputWrapper>
      <div id="ssr-search-input-popout-container">
        <SSRformContainer id="ssr-form-container">
          <form method="get" action={advancedSearchUrl}>
            <SSRsearchInputContainer id="ssr-input-container">
              <SSRmobileSearchIconWrapper>
                <LargeSearchIcon />
              </SSRmobileSearchIconWrapper>
              <SSRdesktopSearchIconWrapper id="ssr-search-icon-wrapper">
                <SearchIcon />
              </SSRdesktopSearchIconWrapper>
              <SSRsearchInputField
                id="ssr-search-input"
                name="text"
                placeholder={placeholder}
                autoComplete="off"
              />
            </SSRsearchInputContainer>
            <script
              data-test-id="ssr-scripts-init"
              ref={scriptRef}
              dangerouslySetInnerHTML={{ __html: script }}
            />
          </form>
        </SSRformContainer>
      </div>
    </SSRsearchInputWrapper>
  );
};
