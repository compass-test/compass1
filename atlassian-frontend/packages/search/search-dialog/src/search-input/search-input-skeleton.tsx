/*
 * Skeleton consumed by product-search-dialog only,
 * check the comments in packages/product-search-dialog/src/common/product-search-input/product-search-input-skeleton.tsx.
 */

import React from 'react';
import { IconButton } from '@atlaskit/atlassian-navigation';
import { useTheme } from '../theme';
import {
  MobileSearchIconWrapper,
  DesktopSearchIconWrapper,
  SearchInputContainer,
  SearchInputField,
} from './search-input.styled';
import { SearchAnchorWrapper } from '../search-anchor';
import { SearchIcon, LargeSearchIcon } from '../icons';
import { getInputSkeletonFocus } from './query-store';
import { SSRsearchInput } from './ssr-interactive-search-input';

export interface Props {
  forwardRef?: React.Ref<HTMLInputElement>;
  features?: {
    interactiveSkeleton?: {
      enabled: boolean;
      advancedSearchUrl: string;
      placeholder: string;
    };
  };
}

const MAX_INPUT_LENGTH = 500;

export const SearchInputSkeleton: React.FunctionComponent<Props> = (props) => {
  const theme = useTheme();
  const { forwardRef, features } = props;

  const { enabled, advancedSearchUrl, placeholder } =
    features?.interactiveSkeleton || {};
  if (enabled || getInputSkeletonFocus()) {
    return (
      <SSRsearchInput
        advancedSearchUrl={advancedSearchUrl}
        placeholder={placeholder}
      />
    );
  }

  return (
    <SearchAnchorWrapper>
      <SearchInputContainer {...theme} isExpanded={false}>
        <MobileSearchIconWrapper isExpanded={false}>
          <IconButton tooltip={null} icon={<LargeSearchIcon />} />
        </MobileSearchIconWrapper>
        <DesktopSearchIconWrapper {...theme} isExpanded={false}>
          <SearchIcon />
        </DesktopSearchIconWrapper>
        <SearchInputField
          data-test-id="search-dialog-search-skeleton"
          isExpanded={false}
          placeholder={placeholder}
          ref={forwardRef || undefined}
          disabled
          value=""
          maxLength={MAX_INPUT_LENGTH}
          onChange={() => {}}
          {...theme}
        />
      </SearchInputContainer>
    </SearchAnchorWrapper>
  );
};

export default React.forwardRef<HTMLInputElement, Props>((props, ref) => (
  <SearchInputSkeleton {...props} forwardRef={ref} />
));
