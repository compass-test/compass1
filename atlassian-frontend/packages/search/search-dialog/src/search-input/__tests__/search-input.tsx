import React from 'react';
import { SearchInput } from '../search-input';
import { SearchInputField, ArrowIconWrapper } from '../search-input.styled';
import { shallow, mount } from 'enzyme';
import { _themes as themes } from '../../theme';

jest.mock('../search-input.styled', () =>
  Object.assign({}, jest.requireActual('../search-input.styled'), {
    HiddenTooltip: () => <div />,
  }),
);

const theme = themes.default;

describe('<SearchInput />', () => {
  it('snapshot with placeholder', () => {
    const wrapper = shallow(
      <SearchInput
        value=""
        isExpanded
        placeholder="placeholder"
        theme={theme}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot expanded', () => {
    const wrapper = shallow(<SearchInput value="" isExpanded theme={theme} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot explicitly collapsed', () => {
    const wrapper = shallow(
      <SearchInput value="" isExpanded={false} theme={theme} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('snapshot with value set', () => {
    const wrapper = shallow(
      <SearchInput value="value is here" isExpanded theme={theme} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('correctly calls onInput when input is entered', () => {
    const callback = jest.fn();
    const wrapper = shallow(
      <SearchInput value="" isExpanded onInput={callback} theme={theme} />,
    );

    wrapper.find(SearchInputField).prop('onInput')!({
      currentTarget: { value: 'a' },
    } as any);

    expect(callback).toBeCalled();
    expect(callback.mock.calls[0][0]).toBe('a');
  });

  it('correctly calls onEnter when enter is pressed', () => {
    const callback = jest.fn();
    const preventDefault = jest.fn();
    const wrapper = shallow(
      <SearchInput value="" isExpanded onEnter={callback} theme={theme} />,
    );

    wrapper.find(SearchInputField).prop('onKeyDown')!({
      preventDefault,
      key: 'Enter',
    } as any);

    expect(callback).toBeCalledTimes(1);
    expect(preventDefault).toBeCalledTimes(1);
  });

  it('does not call onEnter when enter is pressed but the event has defaultPrevented', () => {
    const callback = jest.fn();
    const preventDefault = jest.fn();
    const wrapper = shallow(
      <SearchInput value="" isExpanded onEnter={callback} theme={theme} />,
    );

    wrapper.find(SearchInputField).prop('onKeyDown')!({
      preventDefault,
      key: 'Enter',
      defaultPrevented: true,
    } as any);

    expect(callback).not.toBeCalled();
    expect(preventDefault).not.toBeCalled();
  });

  it('does not render more than 500 characters', () => {
    const spy = jest.spyOn(console, 'warn');
    spy.mockImplementation(() => {});

    const wrapper = mount(<SearchInput value="" isExpanded theme={theme} />);

    expect(
      (wrapper.find(SearchInputField).getDOMNode() as HTMLInputElement)
        .maxLength,
    ).toBe(500);

    spy.mockReset();
  });

  it('calls close when the back button is pressed', () => {
    const onBack = jest.fn();
    const wrapper = shallow(
      <SearchInput value="" isExpanded theme={theme} onBack={onBack} />,
    );

    wrapper.find(ArrowIconWrapper).first().simulate('click');
    expect(onBack).toHaveBeenCalled();
  });
});
