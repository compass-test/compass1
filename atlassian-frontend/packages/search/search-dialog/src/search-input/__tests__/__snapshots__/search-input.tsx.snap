// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> snapshot expanded 1`] = `
<Tooltip
  component={[Function]}
  position="bottom"
>
  <Styled(div)
    default={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#DFE1E6",
        "color": "#6B778C",
        "iconColor": "#344563",
      }
    }
    focus={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#4C9AFF",
        "color": "#091E42",
        "iconColor": "#091E42",
      }
    }
    hover={
      Object {
        "backgroundColor": "rgba(222, 235, 255, 0.9)",
        "borderColor": "#DFE1E6",
        "color": "#0052CC",
        "iconColor": "#0052CC",
      }
    }
    isExpanded={true}
  >
    <Styled(span)
      isExpanded={true}
    >
      <ForwardRef
        icon={<Styled(SearchIcon) />}
        onClick={[Function]}
        tooltip={null}
      />
    </Styled(span)>
    <Styled(span)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <SearchIcon />
    </Styled(span)>
    <Styled(div)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
    >
      <ArrowLeftIcon
        label="Exit Search"
        primaryColor="#6B778C"
        size="large"
      />
    </Styled(div)>
    <Styled(input)
      data-test-id="search-dialog-input"
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      maxLength={500}
      onChange={[Function]}
      onClick={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </Styled(div)>
</Tooltip>
`;

exports[`<SearchInput /> snapshot explicitly collapsed 1`] = `
<Tooltip
  component={[Function]}
  position="bottom"
>
  <Styled(div)
    default={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#DFE1E6",
        "color": "#6B778C",
        "iconColor": "#344563",
      }
    }
    focus={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#4C9AFF",
        "color": "#091E42",
        "iconColor": "#091E42",
      }
    }
    hover={
      Object {
        "backgroundColor": "rgba(222, 235, 255, 0.9)",
        "borderColor": "#DFE1E6",
        "color": "#0052CC",
        "iconColor": "#0052CC",
      }
    }
    isExpanded={false}
  >
    <Styled(span)
      isExpanded={false}
    >
      <ForwardRef
        icon={<Styled(SearchIcon) />}
        onClick={[Function]}
        tooltip={null}
      />
    </Styled(span)>
    <Styled(span)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={false}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <SearchIcon />
    </Styled(span)>
    <Styled(input)
      data-test-id="search-dialog-input"
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={false}
      maxLength={500}
      onChange={[Function]}
      onClick={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </Styled(div)>
</Tooltip>
`;

exports[`<SearchInput /> snapshot with placeholder 1`] = `
<Tooltip
  component={[Function]}
  position="bottom"
>
  <Styled(div)
    default={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#DFE1E6",
        "color": "#6B778C",
        "iconColor": "#344563",
      }
    }
    focus={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#4C9AFF",
        "color": "#091E42",
        "iconColor": "#091E42",
      }
    }
    hover={
      Object {
        "backgroundColor": "rgba(222, 235, 255, 0.9)",
        "borderColor": "#DFE1E6",
        "color": "#0052CC",
        "iconColor": "#0052CC",
      }
    }
    isExpanded={true}
  >
    <Styled(span)
      isExpanded={true}
    >
      <ForwardRef
        icon={<Styled(SearchIcon) />}
        onClick={[Function]}
        tooltip={null}
      />
    </Styled(span)>
    <Styled(span)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <SearchIcon />
    </Styled(span)>
    <Styled(div)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
    >
      <ArrowLeftIcon
        label="Exit Search"
        primaryColor="#6B778C"
        size="large"
      />
    </Styled(div)>
    <Styled(input)
      data-test-id="search-dialog-input"
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      maxLength={500}
      onChange={[Function]}
      onClick={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      placeholder="placeholder"
      value=""
    />
  </Styled(div)>
</Tooltip>
`;

exports[`<SearchInput /> snapshot with value set 1`] = `
<Tooltip
  component={[Function]}
  position="bottom"
>
  <Styled(div)
    default={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#DFE1E6",
        "color": "#6B778C",
        "iconColor": "#344563",
      }
    }
    focus={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#4C9AFF",
        "color": "#091E42",
        "iconColor": "#091E42",
      }
    }
    hover={
      Object {
        "backgroundColor": "rgba(222, 235, 255, 0.9)",
        "borderColor": "#DFE1E6",
        "color": "#0052CC",
        "iconColor": "#0052CC",
      }
    }
    isExpanded={true}
  >
    <Styled(span)
      isExpanded={true}
    >
      <ForwardRef
        icon={<Styled(SearchIcon) />}
        onClick={[Function]}
        tooltip={null}
      />
    </Styled(span)>
    <Styled(span)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <SearchIcon />
    </Styled(span)>
    <Styled(div)
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
    >
      <ArrowLeftIcon
        label="Exit Search"
        primaryColor="#6B778C"
        size="large"
      />
    </Styled(div)>
    <Styled(input)
      data-test-id="search-dialog-input"
      default={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#DFE1E6",
          "color": "#6B778C",
          "iconColor": "#344563",
        }
      }
      focus={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#4C9AFF",
          "color": "#091E42",
          "iconColor": "#091E42",
        }
      }
      hover={
        Object {
          "backgroundColor": "rgba(222, 235, 255, 0.9)",
          "borderColor": "#DFE1E6",
          "color": "#0052CC",
          "iconColor": "#0052CC",
        }
      }
      isExpanded={true}
      maxLength={500}
      onChange={[Function]}
      onClick={[Function]}
      onInput={[Function]}
      onKeyDown={[Function]}
      value="value is here"
    />
  </Styled(div)>
</Tooltip>
`;
